### ISSUE TITLE
E0 qualification bootstrap on Windows

### ISSUE THREAD

**Issue by Alajmah** [ElephantRock/ExpertForge-synthesis#1]
## Authority and source of truth
GitHub is the North Star for this work. The local Windows repository is a working copy only. Do not treat unpushed local state as authoritative.

Authoritative repository: `ElephantRock/ExpertForge-synthesis`
Working branch: `e0/qualification-bootstrap`
Base branch: `main`

## Scope
This issue authorizes only the bounded E0 qualification/bootstrap work already approved in the E0 v0.5 candidate governance flow. It does **not** authorize terminal E0 evidence execution, A0 claims, final corpus generation, or general platform implementation.

## Current authoritative state
- Q1 CMDR qualification surface: **PASS**
- Q2 M0 qualification: **CLOSED — UNQUALIFIED**
  - C0 BELOW_FLOOR
  - C1 BELOW_FLOOR
  - C2 BELOW_FLOOR
  - no C3 exists; no M0 selected
- Q2 terminal evidence: `9990f8c18f14c93a5a40eb19aa4f5e7dd1346399`
- Q2 closure ruling: `docs/experiments/e0/Q2_TERMINAL_CLOSURE.md` at `604ebfe441e79125bd068290f74b8b117a1b0145`
- Q3 tokenizer/interface/runtime prechecks: PASS evidence exists; capability/headroom and source selection are **HELD** because no selected/qualified Arm-A M0 binding exists
- Q4 final 108k corpus: **NOT AUTHORIZED**
- Q5/Q6 and pre-freeze binding snapshot: **BLOCKED**
- E0 v0.5 remains **DRAFT**; FROZEN is not reachable from the present qualification outcome

## Disposition
The frozen M0 ladder C0→C1→C2 was executed exactly as registered across the three frozen seeds and 8,000 updates per seed. All three candidates remained below the Decision-6 floor at chance-level qualification performance. Q2 therefore closes as `UNQUALIFIED`. This is a failure of the prerequisite qualification path, not evidence against A0 or the Expertise Adapter hypothesis, because adapter utility was never tested.

Do not invent C3, retune v0.5 after seeing the result, reinterpret C2 as a qualified Arm-A pilot, or generate the final corpus. Any additional scientific work belongs in a separate diagnostic/revision lane and requires a new contract candidate before qualification can resume.

## Preserved operating rules
- Never force-push this branch after shared work begins.
- Never commit secrets, access tokens, model-provider credentials, or machine-private identifiers.
- Do not commit downloaded model weights. Record immutable upstream revision/digest metadata instead.
- Do not modify frozen thresholds, seeds, arm definitions, data rules, optimizer rules, or statistical rules to accommodate observed results.
- If local execution contradicts a contract or integrity invariant, stop and report before changing scientific configuration.
- `main` is accepted project state; this issue + branch + draft PR #2 are the working coordination surface.

**Comment by Alajmah** [comment]
Bootstrap coordination surface is ready on branch `e0/qualification-bootstrap`.

Windows assistant: pull the branch and begin with `docs/experiments/e0/LOCAL_ASSISTANT_HANDOFF.md`. Run `scripts/e0/bootstrap_windows.ps1` first. Do **not** start Q2/Q3 until the generated environment snapshot and local Q1 reproduction are committed and pushed here.

Current branch tip after bootstrap scaffolding: `27d0115b7577be22588adf09667f87c10a8afa99`.

**Comment by Alajmah** [comment]
Updated bootstrap branch now includes the exact repo-local Q1 generator, independent verifier/audits, pinned Q1 dependencies, and a Windows Q1 runner. Current branch tip: `c4e9a4543cb210f600463458d207c3432e2e1d01`.

Windows sequence:
```powershell
git fetch origin
git checkout e0/qualification-bootstrap
git pull --ff-only origin e0/qualification-bootstrap
powershell -ExecutionPolicy Bypass -File .\scripts\e0\bootstrap_windows.ps1
powershell -ExecutionPolicy Bypass -File .\scripts\e0\run_q1_windows.ps1
```
If Q1 is not exact PASS, stop and report here. If it passes, commit/push `docs/experiments/e0/windows_environment.snapshot.json`, `docs/experiments/e0/q1_local_reproduction.json`, and `docs/experiments/e0/LOCAL_BOOTSTRAP_STATUS.md`, then comment with the commit SHA and GPU/RAM summary. Do not begin Q2/Q3 until that push/report is complete.

**Comment by Alajmah** [comment]
Local Windows bootstrap report received. Q1 reproduced exactly: PASS. Environment summary reported locally: Windows 11 Pro build 26200, Ryzen 5 5600G, ~31.9 GB RAM, RTX 3080 Ti 12 GB, NVIDIA driver 616.64, Python 3.12.10. A local commit `eca842f` was created containing the first deliverable but has not yet been pushed, so GitHub does not yet have authority over that state.

Next action for the local assistant:

```powershell
git status
git log -1 --oneline
git push origin e0/qualification-bootstrap
```

After the push succeeds, report the full pushed commit SHA here. Do not start Q2/Q3 until the pushed branch tip is visible on GitHub and this issue records the commit. Keep the first pushed deliverable limited to the environment snapshot, Q1 reproduction summary/status, and minimal bootstrap scripts already authorized.

**Comment by Alajmah** [comment]
Verified the first Windows commit directly on GitHub: `eca842fad54a52db638dd913189ef9ada1dc69a6` contains exactly the expected three deliverables, and Q1 reports PASS with all four frozen corpus hashes matching. Draft PR #2 is now open against `main`.

I also corrected one resource interpretation before Q2/Q3: Decision 25's 12 GiB accelerator / 64 GiB host / 128 GiB workspace values are ceilings, not minimum installed capacities. The Windows host's ~32 GiB RAM therefore does not fail qualification by itself; measured execution simply must fit physical memory and stay within the frozen ceilings.

New authoritative branch head includes:
- `docs/experiments/e0/S0_CANDIDATE_REGISTRY_FROZEN.yaml`
- `docs/experiments/e0/Q2_Q3_EXECUTION_RELEASE.md`
- `scripts/e0/requirements_q2_q3.txt`

Current head after those additions: `304f19d04f22f0ff8aff370498831fc5bffcc6b5`.

## Next local step
Pull the branch, read `Q2_Q3_EXECUTION_RELEASE.md`, install the pinned Q2/Q3 runtime, update the runtime snapshot, then implement the Q2/Q3 harnesses and run only short smoke tests. Push that harness/smoke-test commit before expensive full qualification. Do **not** run the final 108k corpus generation.

Q2 full qualification is released after that harness commit; Q3 tokenizer/interface/runtime prechecks may be implemented in parallel, but Q3 capability/headroom cannot close until Q2's three Arm-A pilot results exist.

**Comment by Alajmah** [comment]
Verified harness/smoke commit `f5d5a88b376f0dba214b2d632947762d9a576124` directly from GitHub and issued the authoritative Q2 full-run ruling in `docs/experiments/e0/Q2_FULL_RUN_RELEASE.md` at commit `1070a61c4f48cef4d02bc6ac6305c31fe09b4117`.

Four interpretation rulings are now closed:
1. **Q2 weight decay:** full scientific qualification must use `exclude_norm_bias` — decay embeddings/attention/MLP/classifier matrix only; RMSNorm scales and classifier bias get zero decay. The current `--wd-scope all` default must not be used for full Q2.
2. **Q2 permutation derivation:** approved exactly as implemented, with zero-based epoch and big-endian first 8 SHA-256 bytes.
3. **Q3 answer context:** exact no-newline `Answer:` suffix followed by literal `" A"`, `" B"`, `" C"`; remove candidate-specific bare-letter fallback.
4. **Q3 replay subset:** approved exactly as implemented: 20 per label×depth cell from eval_ID by ascending SHA-256 rank under `ExpertForge-E0-Q3|replay|<sample_id>`.

Before any full 8,000-update run, pull `1070a61c...`, patch Q2/Q3 accordingly, rerun Q2 smoke + 16-update path under `exclude_norm_bias`, rerun Q3 interface smoke with exact leading-space continuation only, update status, commit and push the correction checkpoint.

After that correction checkpoint passes, **C0 full Q2 qualification is authorized** for the three frozen seeds only. Push the complete C0 evidence before deciding whether the frozen ladder proceeds to C1. Do not generate the final 108k corpus.

**Comment by Alajmah** [comment]
GitHub-side validation of correction checkpoint `aec481327e8e73467662f5469bcb5c50cbaf480a` confirms R1/R3 were implemented correctly and corrected smokes PASS. However, prelaunch audit found four implementation-integrity blockers before the first scientific 8,000-update Q2 run: (P1) re-verify frozen Q1 input SHA-256s at scientific startup, (P2) fail closed on any non-finite scientific divergence so an earlier best checkpoint cannot rescue the seed, (P3) fix stale summary metadata that still says weight-decay default is `all` and record code/data/runtime lineage in scientific results, and (P4) regenerate smoke manifests from a clean committed code revision because current smoke files say `git_commit=1070a61...` although they were produced with uncommitted R1/R3 changes. Full details are now authoritative in `docs/experiments/e0/Q2_PRELAUNCH_INTEGRITY_AUDIT.md`, added at branch commit `d03ae14c675fce31c4d7b18aec08bfb4860ef08f`. C0 full Q2 is HELD until P1-P4 close. This is an implementation/evidence correction only; no scientific design rule changes.

**Comment by Alajmah** [comment]
GitHub-side validation complete for the Q2 prelaunch integrity checkpoints.

Verified code-fix commit A: `35c6a8f21169202dbc297fa5908f5f09d6c7d39c` closes P1–P3 (frozen Q1 input re-hash, fail-closed divergence handling, corrected scientific lineage metadata).

Verified evidence-refresh commit B / current branch head: `48b2597f14405ea046c71e627a5bc63a54feb85a`. The refreshed Q2/Q3 smoke manifests all PASS and bind to clean code revision `35c6a8f...` as required by P4. PR #2 currently points to `48b2597f...`.

**Release:** C0 full Q2 qualification is now authorized from exact branch head `48b2597f14405ea046c71e627a5bc63a54feb85a` only. Run `--candidate C0`, all three frozen seeds, all 8,000 updates, under `exclude_norm_bias`; do not shorten or branch on intermediate validation. After completion, push the three per-seed scientific results plus `q2_m0_qualification_summary.json` before any C1 decision.

Q3 capability/headroom remains blocked until those C0 Arm-A results are pushed. Final 108k corpus generation remains unauthorized.

**Comment by Alajmah** [comment]
C0 launch infrastructure incident recorded. The first full-run invocation at `48b2597f14405ea046c71e627a5bc63a54feb85a` failed during scientific entry-path lineage assembly because `subprocess.run(...)` was referenced without a module-level `subprocess` import. GitHub code inspection confirms that call occurs before the seed loop/model construction.

Classification: `INFRASTRUCTURE_INVALID_PRE_MODEL`. No seed entered training, no update/metric/checkpoint/result was produced, and this is not a scientific/algorithmic failure.

Authoritative remediation protocol is now in `docs/experiments/e0/Q2_C0_INFRASTRUCTURE_RETRY_RELEASE.md` at branch commit `979cacbe0a372fbb532e4441ceb57aadfd88823a`.

C0 is HELD pending two commits:
- Commit C: import fix + shared scientific setup + new `--preflight` + machine-readable incident record binding preserved local logs by SHA-256.
- Commit D: from clean C, PASS `py_compile`, exact C0 scientific preflight, Q2 smoke, and 16-update non-scientific path; evidence manifests must bind to C.

Q3 need not be rerun for this Q2-only defect. No scientific settings may change. Do not relaunch C0 until GitHub validates C and D.

**Comment by Alajmah** [comment]
Validated remediation commits C and D directly on GitHub.

Accepted code commit C: `7d0ebd9ec437c4d86c962faba021ffb3b63a00b4`.
Accepted evidence commit D / current branch head: `7556f756146e59940ada037385c81fdf5325bc1e`.

Findings:
- the launcher incident record is complete and correctly classified `INFRASTRUCTURE_INVALID_PRE_MODEL` with zero seeds/updates/model construction;
- module-level `subprocess` import is present;
- `scientific_setup()` is shared by both full `--candidate` execution and `--preflight` and covers wd-scope rejection, deterministic controls, CUDA gate, frozen-Q1 rehash, lineage assembly, data load, exact candidate construction, and AdamW parameter groups;
- preflight PASS is bound to clean code commit C, first frozen C0 seed `1647674144`, exact Q1 digests, C0 parameter count `53,232,643`, and Decision-22 split `53,216,768` decay / `15,875` no-decay;
- refreshed Q2 smoke PASS is also bound to commit C;
- PR #2 currently points to evidence commit D.

Disposition: **C0 INFRASTRUCTURE RETRY RELEASED** from exact branch head `7556f756146e59940ada037385c81fdf5325bc1e`, provided the local tree is clean and HEAD matches exactly at launch. The retry uses the original frozen C0 protocol: all three frozen seeds, 8,000 updates each, `exclude_norm_bias`. The pre-model incident does not count as a scientific seed attempt.

Do not start C1 or Q3 capability scoring. After C0 completes, push the three scientific seed result records plus `q2_m0_qualification_summary.json` before applying the frozen ladder.

**Comment by Alajmah** [comment]
Independent Decision-6 recomputation from pushed C0 evidence commit `226b670a9fe59f94395a3db995c67a1a2e2145c2`:

Per-seed Q = [0.3333333333333333, 0.3333333333333333, 0.33341666666666664]. Median Q = 0.3333333333333333, median Q2:4 = 0.3333333333333333, median Q_STRUCT = 0.3333333333333333, median min-label-recall = 0.0. All five frozen learnability checks fail. C0 is not above the 0.70 ceiling, so the frozen ladder classifies C0 as BELOW_FLOOR and **C1 is now eligible**. This is not yet Q2-UNQUALIFIED; that outcome is only possible after the final C2 candidate also fails.

Release: run C1 only, all three frozen seeds, all 8,000 updates, unchanged scientific recipe and fail-closed rules. Do not start C2, Q3 capability scoring, or final corpus generation. After C1 completes, push all three C1 seed records plus updated `q2_m0_qualification_summary.json` before any further ladder decision.

**Comment by Alajmah** [comment]
Decision-6 recomputed independently from C1 evidence commit `12bbe9e888ba1c1450ea6720ad80da9b4a628718`. All three frozen C1 seeds completed 8,000 updates under the frozen recipe with Q = 0.3333333333333333 each. The pushed summary gives median Q = 0.3333333333333333, median Q2:4 = 0.3333333333333333, median Q_STRUCT = 0.3333333333333333, median min-label-recall = 0.0, and every-seed Q-in-band = false. C1 is below the 0.55 learnability floor and not above the 0.70 ceiling. **C2 is released** as the final frozen M0 candidate: same three seeds, 8,000 updates each, unchanged optimizer/data/checkpoint recipe. No C3 exists. If C2 also fails Decision-6, Q2 closes UNQUALIFIED and no Q3 capability/headroom or final-corpus generation should proceed under this candidate contract until the project authority determines the next contract version.

**Comment by Alajmah** [comment]
Q2 terminal closure validated from evidence commit `9990f8c18f14c93a5a40eb19aa4f5e7dd1346399` and recorded in `docs/experiments/e0/Q2_TERMINAL_CLOSURE.md` at branch head `604ebfe441e79125bd068290f74b8b117a1b0145`.

Ruling: **Q2 = UNQUALIFIED**. C0/C1/C2 all fail Decision-6 below floor; no C3 exists and no M0 is selected. This is a prerequisite-qualification failure, not an adapter/A0 result.

Downstream: Q3 interface/runtime prechecks remain evidence, but Q3 capability/headroom and source selection are held because there is no selected/qualified Arm-A M0 binding. Do not repurpose C2 as the selected Arm-A reference. Final 108k corpus is not authorized; Q5/Q6 and pre-freeze bindings remain blocked; E0 v0.5 stays DRAFT and cannot advance to FROZEN.

Any further work must be a separate non-qualification diagnostic/revision lane leading to a new contract candidate; v0.5 evidence must remain immutable.
